lib/internal/errors.js

```diff
@@ -194,7 +194,49 @@ function E(sym, val) {
   messages.set(sym, typeof val === 'function' ? val : String(val));
 }
 
+// JS counterpart of StringFromPath, although here path is a buffer.
+function stringFromPath(path) {
+  const str = path.toString();
+  if (process.platform !== 'win32') {
+    return str;
+  }
+
+  if (str.startsWith('\\\\?\\UNC\\')) {
+    return '\\\\' + str.slice(8);
+  } else if (str.startsWith('\\\\?\\')) {
+    return str.slice(4);
+  }
+  return str;
+}
+
+// This creates an error compatible with errors produced in UVException
+// using the context collected in CollectUVExceptionInfo
+// The goal is to migrate them to ERR_* errors later when
+// compatibility is not a concern
+function uvException(ctx) {
+  const err = new Error();
+  err.errno = ctx.errno;
+  err.code = ctx.code;
+  err.syscall = ctx.syscall;
+
+  let message = `${ctx.code}: ${ctx.message}, ${ctx.syscall}`;
+  if (ctx.path) {
+    const path = stringFromPath(ctx.path);
```

Comment:
Also @fhinkel does this still LGTY with the `stringFromPath` changes for fixing Windows?

Reply:
Still LGTM